ci(sgv-08): commit standardization tool

matheusbrs opened the pull request:
<h2>Installation of the following tools for commit standardization</h2>
<ul>
<li>Husky</li>
<li>Commitlint</li>
<li>Commitizen</li>
</ul>

<p>
Husky, Commitlint and Commitizen help standardize commits. Husky runs automatic checks before committing, Commitlint ensures messages follow a pattern, and Commitizen makes it easy to create consistent messages. Together, these tools organize and improve commit history.
</p>
<h5>OBSERVAÇÃO</h5>
<p>
The text "(SGV08)" was automatically formatted to lowercase when pressing Enter.
 </p>

matheusbrs committed:
ci(sgv-08): commit standardization tool

MatheusMFranco reviewed (changes requested):
Antes de mais nada, exclua a pastanode_modules, crie um arquivo chamado .gitignore e coloque a pasta node_modules neste arquivo, assim vai ignorar ela na subida. Esta pasta são os pacotes instalados, a gente não precisa dela no repositório, porque o proprio dev consegue gerar ela quando instala os pacotes.

matheusbrs committed:
chore(sgv-08): adds .gitignore to ignore the node_modules folder

This commit creates a .gitignore file that prevents the node_modules folder, which contains project
dependencies, from being included in the Git repository. This helps keep the repository clean and
lightweight, as the node_modules folder can be easily regenerated with the 'npm install' command.

MatheusMFranco commented on .husky/pre-commit:
Ao invés de `pre-commit`, tem que ser `commit-msg`, porque você vai validar a *mensagem*.
Além disso você precisa chamar o `commitlint` no arquivo, até então ele não está sendo chamado.

MatheusMFranco reviewed (changes requested):
Você tá usando o yarn, npm e pnpm ao mesmo tempo? Escolha só um. 🤣

matheusbrs committed:
chore(sgv08): setup of Husky, Commitizen, and Commitlint for standardized commit control

Setting up Husky, Commitizen, and Commitlint to enforce standardized commit messages across the
project. This configuration integrates Husky hooks to automate the commit message validation
process, Commitizen to provide an interactive prompt for creating conventional commits, and
Commitlint to check that all commit messages follow a specific format. Together, these tools ensure
consistent and readable commit history, improving collaboration and simplifying project maintenance.

MatheusMFranco commented on .commitlintrc.js:
```diff
@@ -1,8 +1,7 @@
     rules: {
-      'scope-enum': [2, 'always', [/^SGV-\d+$/]],
-      'scope-empty': [2, 'never'], 
+      'scope-enum': [2, 'always', [/^SGV-\d+$/]], // Valida o padrão SGV-NÚMERO com uma expressão regular, sendo "d" dígitos decimais
```
Tira os comentários.

MatheusMFranco replied:
Tira o traço da expressão regular.

MatheusMFranco reviewed (changes requested):
Remove os comentários que vai estar show.

matheusbrs committed:
Testando commitlint

matheusbrs committed:
chore(sgv08): setup of Husky, Commitizen, and Commitlint for standardized commit control

matheusbrs committed:
Testando commitlint

matheusbrs committed:
Testando commitlint

matheusbrs committed:
chore: add eslint config

matheusbrs committed:
chore(sgv08): removes the dash from the regular expression

matheusbrs committed:
feat: Ajuste da pagina
